src/components/slide-toggle/slide-toggle.spec.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  it,
+  describe,
+  expect,
+  beforeEach,
+  inject,
+  async,
+} from '@angular/core/testing';
+import {TestComponentBuilder, ComponentFixture} from '@angular/compiler/testing';
+import {By} from '@angular/platform-browser';
+import {Component} from '@angular/core';
+import {MdSlideToggle} from './slide-toggle';
+
+export function main() {
+  describe('MdSlideToggle', () => {
+    let builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+      builder = tcb;
+    }));
+
+    describe('basic behavior', () => {
+
+      let fixture: ComponentFixture<any>;
+
+      let testComponent: SlideToggleTestApp;
+      let slideToggle: MdSlideToggle;
+      let slideToggleElement: HTMLElement;
+      let labelElement: HTMLLabelElement;
+      let inputElement: HTMLInputElement;
+
+      beforeEach(async(() => {
+        builder.createAsync(SlideToggleTestApp).then(f => {
+          fixture = f;
+          fixture.detectChanges();
+
+          testComponent = fixture.debugElement.componentInstance;
+          slideToggle = fixture.debugElement.query(By.css('md-slide-toggle')).componentInstance;
+          slideToggleElement = fixture.debugElement.query(By.css('md-slide-toggle')).nativeElement;
+          inputElement = fixture.debugElement.query(By.css('input')).nativeElement;
+          labelElement = fixture.debugElement.query(By.css('label')).nativeElement;
+        });
+      }));
+
+
+      it('should update the model correctly', () => {
+        expect(slideToggleElement.classList).not.toContain('md-checked');
+
+        testComponent.slideModel = true;
+        fixture.detectChanges();
+
+        expect(slideToggleElement.classList).toContain('md-checked');
+      });

```

Comment:
There should be a test checking that the `NgControl` pristine and touched states are set correctly. 
You can see where I do this for radio button in my PR: https://github.com/angular/material2/pull/443/files#diff-ebc684a852af87cf7ac222fa6411cbe2R228